packages/xo-server/src/api/xosan.js

```diff
@@ -1070,7 +1070,7 @@ export const addBricks = defer(async function ($defer, { xosansr, lvmsrs, brickS
       data.type = 'replica'
       await xapi.xo.setData(xosansr, 'xosan_config', data)
       await glusterCmd(glusterEndpoint, 'peer detach ' + arbiterNode.vm.ip, true)
-      await xapi.deleteVm(arbiterNode.vm.id, true)
+      await xapi.VM_destroy(arbiterNode.vm.$ref, true)
```

Comment:
`arbiterNode.vm` isn't an XAPI object.

Reply:
Indeed, it appears to be an VM UUID, you need to resolve it:

```suggestion
      await xapi.VM_destroy(await xapi.call('VM.get_by_uuid', arbiterNode.vm.id), true)
```

Reply:
In this case we are not sending to `VM destroy` a `ref`. 
`VM_destroy` should handle the case of non `ref` parameter ?

Reply:
`await xapi.call('VM.get_by_uuid', arbiterNode.vm.id)` will return the ref corresponding to the UUID.